src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -69,47 +73,115 @@ IAsyncServiceProvider serviceProvider
             IProjectSystemCache projectSystemCache,
             ISolutionRestoreWorker restoreWorker,
             ILogger logger,
-            Microsoft.VisualStudio.Threading.AsyncLazy<IVsSolution2> vsSolution2)
+            Microsoft.VisualStudio.Threading.AsyncLazy<IVsSolution2> vsSolution2,
+            INuGetTelemetryProvider telemetryProvider)
         {
             _projectSystemCache = projectSystemCache ?? throw new ArgumentNullException(nameof(projectSystemCache));
             _restoreWorker = restoreWorker ?? throw new ArgumentNullException(nameof(restoreWorker));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _vsSolution2 = vsSolution2 ?? throw new ArgumentNullException(nameof(vsSolution2));
+            _telemetryProvider = telemetryProvider ?? throw new ArgumentNullException(nameof(telemetryProvider));
         }
 
-        public Task<bool> CurrentRestoreOperation => _restoreWorker.CurrentRestoreOperation;
+        Task<bool> IVsSolutionRestoreService.CurrentRestoreOperation
+        {
+            get
+            {
+                const string eventName = nameof(IVsSolutionRestoreService) + "." + nameof(IVsSolutionRestoreService.CurrentRestoreOperation);
+                NuGetExtensibilityEtw.EventSource.Write(eventName, NuGetExtensibilityEtw.InfoEventOptions);
```

Comment:
Can we consider using VS ETW library ?

https://www.fuget.org/packages/Microsoft.VisualStudio.Utilities/17.0.0-preview-2-31221-277/lib/netstandard2.0/Microsoft.VisualStudio.Utilities.dll/Microsoft.Internal.VisualStudio.Diagnostics/VsEtwLogging

See `public static void WriteEvent<T>(string eventName, VsEtwKeywords keyword, VsEtwLevel level, T data, Guid relatedActivityId)` or related method overloads.

Reply:
I addressed this in the PR description (although I needed to edit it a little, as it had some bad edits making it difficult to understand.

> I chose to create one specifically for this feature because we listen to it with EventListener to report call counts, and I thought it's better to not have to filter out a large number of events using more generic ETW providers, particularly the VS common provider.